packages/core/src/internal/execution/future-processor/helpers/future-resolvers.ts

```diff
@@ -117,6 +116,25 @@ export function resolveFutureFrom(
   return resolveAccountRuntimeValue(from, accounts);
 }
 
+/**
+ * Resolve a `send` future's data parameter to a string.
+ */
+export function resolveFutureData(
+  data: string | EncodeFunctionCallFuture<string, string> | undefined,
+  deploymentState: DeploymentState
+): string {
+  if (data === undefined) {
+    return "0x";
+  }
+
+  if (typeof data === "string") {
+    return data;
+  }
+
+  // this type coercion is safe because we know the type of data is EncodeFunctionCallFuture
+  return findResultForFutureById(deploymentState, data.id) as string;
```

Comment:
Lets initially return a `result` and then do a `assertIgnitionInvariant` to ensure it is a string.